datalad/support/annexrepo.py

```diff
@@ -2124,7 +2124,9 @@ def _run_annex_command_json(self, command,
             if e.stderr:
                 # else just warn about present errors
                 shorten = lambda x: x[:1000] + '...' if len(x) > 1000 else x
-                lgr.warning(
+
+                _log = lgr.debug if kwargs.get('expect_stderr', False) else lgr.warning
```

Comment:
In deciding to log at 9 or 11, `Runner` looks at `expect_fail` rather than `expect_stderr` in the case of a non-zero exit status.  Given that we're processing a non-zero exit (we're under `except CommandError`), should we look at `expect_fail` rather than `expect_stderr`?

Reply:
You're absolutely right. Will change.

Reply:
Although - it is somewhat confusing how exactly we want to have it. `expect_stderr` and `expect_fail` are different, independent parameters. So, what should be the meaning of combining them? If I declare to expect a non-zero exit, but not expecting something on stderr shouldn't we still have it logged as a warning? I guess, we are not really consistent in the interpretation of those.
However, I agree and would like to go with the simple approach: really consider `expect_stderr` only for zero exit.

Reply:
> Although - it is somewhat confusing how exactly we want to have it. `expect_stderr` and `expect_fail` are different, independent parameters. So, what should be the meaning of combining them? If I declare to expect a non-zero exit, but not expecting something on stderr shouldn't we still have it logged as a warning?

Yeah, I find the interaction between `expect_stderr` and `expect_fail` confusing.  I can find spots in cmd.py where it clearly looks at `expect_fail` if the status is non-zero and `expect_stderr` otherwise, but I don't think that's necessarily the behavior I'd expect from their docstring descriptions.  I'd guess we could collapse them into one parameter and retain all the cases we actually care about.

Anyway, for the case of this change, where we are in a non-zero exit, I think whether we look at `expect_stderr` or `expect_fail` isn't a big deal because I'd think `expect_fail == expect_stderr` would be the dominant case.